test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/SelectExpandFunctionalTests.cs

```diff
@@ -1414,6 +1414,77 @@ public void DollarItinFilterInsideSelectShouldReferenceResourcePathEntity()
             Assert.Equal(expectedLeft, left);
         }
 
+        [Theory]
+        [InlineData("RelatedSSNs($orderby=$this)", OrderByDirection.Ascending)]

```

Comment:
Do you have any $this tests for collections of entities and complex types? i.e.:
People/{id}/PreviousAddresses?$filter=$this/Street eq 'abc'
or
People?$expand=friends($filter=contains($this/lastName,'wilson'))

While it's not required to explicitly use $this in these cases, it should still be supported.

Reply:
I have added the test.
#Resolved